x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/split_field/split_field.tsx

```diff
@@ -23,7 +26,11 @@ export const SplitFieldSelector: FC = () => {
   const jobCreator = jc as MultiMetricJobCreator | PopulationJobCreator;
   const canClearSelection = isMultiMetricJobCreator(jc);
 
-  const { categoryFields } = newJobCapsService;
+  const [runtimeCategoryFields] = useState(filterCategoryFields(jobCreator.runtimeFields));
+  const [categoryFields] = useState([
+    ...newJobCapsService.categoryFields,
+    ...runtimeCategoryFields,
+  ]);
```

Comment:
Does this need `useState` if a `setFields` is never needed? Maybe it doesn't need to be wrapped in `useState` at all or can make use of `useMemo`?

Reply:
updated in https://github.com/elastic/kibana/pull/91168/commits/3e19ad9587a3e70040dcf4523c0e377a0bd54ae2